tests/Doctrine/Tests/PHPUnitCompatibility/ExceptionMatching.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\PHPUnitCompatibility;
+
+use function get_parent_class;
+use function method_exists;
+
+trait ExceptionMatching
+{
+    /**
+     * Override for BC with PHPUnit <8
+     */
+    public function expectExceptionMessageMatches(string $regularExpression): void
+    {
+        if (method_exists(get_parent_class($this), 'expectExceptionMessageMatches')) {
+            parent::expectExceptionMessageMatches($regularExpression);
+        } else {
+            parent::expectExceptionMessageRegExp($regularExpression);
+        }
+    }
+}
```

Comment:
I think this should go in there: https://github.com/doctrine/orm/blob/3843d7e0cc71c78c359af4f962b5ec2efc0312cd/tests/Doctrine/Tests/DoctrineTestCase.php#L20-L24

Reply:
Wasn't aware of that – updated.